plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -589,6 +598,13 @@ abstract class Rule {
      */
     public abstract RuleType type();
 
+    /**
+     * @see NewRule#setCharacteristic(RuleCharacteristic)
+     * @since 9.16
+     */
+    @CheckForNull
+    public abstract RuleCharacteristic ruleCharacteristic();
```

Comment:
I'm not sure if this should be null.
Even if we have a default value, it should probably be returned at this point

Reply:
As discussed, we won't have default values in the API